ui/components/MeshConfigComponent.js

```diff
@@ -232,7 +235,8 @@ class MeshConfigComponent extends React.Component {
   setOperatorState = (res) => {
     const self = this;
     if (res.operator?.error) {
-      self.handleError("Operator could not be reached")(res.operator?.error?.description)
+      if (this.state.isMeshConfigured)
```

Comment:
I am not exactly clear about the intentions behind this, can you explain ?

Reply:
> I am not clear about the intentions behind this, can you explain ?

This also checks for the status of the metrics through line 216
```javascript
componentDidMount() {
    const self = this;
    // Subscribe to the operator events
    subscribeMeshSyncStatusEvents((res) => {
      if (res.meshsync?.error) {
        self.handleError(res.meshsync?.error?.description || "MeshSync could not be reached");
        return;
      }
    });

    subscribeOperatorStatusEvents(self.setOperatorState);
    fetchMesheryOperatorStatus().subscribe({
      next: (res) => {
        self.setOperatorState(res);
      },
      error: (err) => console.log("error at operator scan: " + err),
    });
  }
```
and it will meet an error and pops up an error message of "Kubernetes is not configured" or something along the lines.

Reply:
From my understanding, `isMeshConfigured` in this file is just checking if the `k8sfile` exists or not.
Now, the function `setOperatorState` is used to set the state of operator based on the result from the graphql query. That is supposed to show an error if the graphql query returned an error, in that case , line 238 introduces a condition that if only `isMeshConfigured` is `true`, that is, if only `k8sfile` is true, the error will be handled, which does not make sense.

Reply:
Now that you say it, that's true. Then maybe this is better
```javascript
if(this.state.isMetricsConfigured){
      subscribeOperatorStatusEvents(self.setOperatorState);
      fetchMesheryOperatorStatus().subscribe({
        next: (res) => {
          self.setOperatorState(res);
        },
        error: (err) => console.log("error at operator scan: " + err),
      });
    }
```
Putting it outside the function so it block sending the request initially if the k8s file isn't there.